### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by Mutombe**
## Summary

This pull request addresses issue #886 by introducing comprehensive tests for five views in the Projects app. These tests ensure the reliability and correctness of the views, enhancing the overall quality and stability of the application.

## Changes
The following changes have been made in this pull request:

1. Implemented tests for the `ProjectCreateAPIView:` These tests cover various scenarios related to project creation, validating inputs, handling permissions, and verifying the appropriate response codes and data.
2. Implemented tests for the `ProjectUpdateAPIView:` These tests thoroughly examine the functionality of updating project details, including edge cases, error handling, and ensuring the proper behavior of the view under different circumstances.
3. Implemented tests for the `ProjectDeleteAPIView:` These tests assess the behavior of the project deletion process, verifying that the appropriate permissions are enforced, confirming the correct response codes, and ensuring the database integrity is maintained.
4. Implemented tests for the `ProjectDetailsAPIView:` These tests comprehensively evaluate the functionality of retrieving project details, covering different scenarios, handling various input cases, and validating the correctness of the returned data.

Each of these test implementations incorporates thorough test coverage to safeguard against regressions and ensure the robustness of the Projects app.

By introducing these tests, we can confidently assert the reliability and stability of these critical views, mitigating potential issues and providing a more professional and polished experience for our users.

Please review the changes and provide your feedback.

### CODE REVIEW

File: zubhub_backend/zubhub/projects/tests.py [hunk 1]
```diff
@@ -1,3 +1,140 @@
...
+            "publish": {"type": 4, "visible_to": []}
+        }
+        request = self.factory.patch(f'/projects/{self.project.pk}/', data)
+        request.user = AnonymousUser()
+
+        response = ProjectUpdateAPIView.as_view()(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+    def test_create_project_unauthenticated_user(self):
```
Comment (NdibeRaymond) [inline zubhub_backend/zubhub/projects/tests.py:83]: this should be in `ProjectCreateAPITest` and not `ProjectUpdateAPITest` right?

**Review by NdibeRaymond — CHANGES_REQUESTED** [review]
